core/src/main/scala/com/nvidia/spark/rapids/tool/analysis/AppSQLPlanAnalyzer.scala

```diff
@@ -65,6 +70,113 @@ class AppSQLPlanAnalyzer(app: AppBase, appIndex: Int) extends AppAnalysisBase(ap
     }
   }
 
+  /**
+   * Update the potential problems in the app object. This is a common function that is used by
+   * Both Qual/Prof analysis.
+   * For Qual apps, the app.sqlIDtoProblematic won't be set because it is done later during the
+   * aggregation phase.
+   * @param sqlId the SQL ID being analyzed
+   * @param potentialProblems a set of strings that represent the potential problems found in the
+   *                          SQL plan.
+   */
+  private def updateAppPotentialProblems(sqlId: Long,
+      potentialProblems: mutable.AbstractSet[String]): Unit = {
+    // Append problematic issues to the global variable for that SqlID
+    val existingIssues = app.sqlIDtoProblematic.getOrElse(sqlId, mutable.LinkedHashSet[String]())
+    app.sqlIDtoProblematic(sqlId) = existingIssues ++ potentialProblems
+    // update the SQLInfoClass with the problematic issues
+    app.sqlIdToInfo.get(sqlId).foreach { sqlInfoClass =>
+      sqlInfoClass.problematic = ToolUtils.formatPotentialProblems(potentialProblems.toSeq)
+    }
+  }
+
+  // A visitor context to hold the state of the SQL plan visitor
+  protected case class SQLPlanVisitorContext(
+      sqlPIGEntry: SqlPlanInfoGraphEntry,
+      sqlDataSources: ArrayBuffer[DataSourceCase] = ArrayBuffer[DataSourceCase](),
+      potentialProblems: collection.mutable.LinkedHashSet[String] = mutable.LinkedHashSet[String](),
+      var sqlIsDsOrRDD: Boolean = false)
+
+  /**
+   * Visit a node in the SQL plan and process:
+   * 1- the metrics;
+   * 2- potential problems.
+   * 3- data sources
+   * @param visitor the visitor context defined per SQLPlan
+   * @param node the node being currently visited.
```

Comment:
no return value so what does it do - ie side effect?  please document

Reply:
Updated the docs